apis/Google.Cloud.Firestore/Google.Cloud.Firestore/DocumentReference.cs

```diff
@@ -119,6 +119,33 @@ public CollectionReference Collection(string path)
             var results = await batch.CommitAsync(cancellationToken).ConfigureAwait(false);
             return results[0];
         }
+        
+        /// <summary>
+        /// Asynchronously performs a set of updates on the document referred to by this path, with an optional precondition.
+        /// </summary>
+        /// <param name="updates">The updates to perform on the document, keyed by the dot-separated field path to update. Fields not present in this dictionary are not updated. Must not be null or empty.</param>
+        /// <param name="precondition">Optional precondition for updating the document. May be null, which is equivalent to <see cref="Precondition.MustExist"/>.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor for the asynchronous operation.</param>
+        /// <returns>The write result of the server operation.</returns>
+        public Task<WriteResult> UpdateAsync(IDictionary<string, object> updates, Precondition precondition = null, CancellationToken cancellationToken = default)
+        {
+            GaxPreconditions.CheckNotNull(updates, nameof(updates));
+            return UpdateAsync(updates.ToDictionary(pair => FieldPath.FromDotSeparatedString(pair.Key), pair => pair.Value), precondition, cancellationToken);
+        }
+
+        /// <summary>
+        /// Asynchronously performs a set of updates on the document referred to by this path, with an optional precondition.
```

Comment:
"a set of updates" => "an update"?

Reply:
Humbug - thanks, will fix. (It's probably wrong in Transaction too.)

Reply:
Well spotted ;)